src/lightning_app/storage/drive.py

```diff
@@ -219,7 +219,7 @@ def to_dict(self):
             "protocol": self.protocol,
             "allow_duplicates": self.allow_duplicates,
             "component_name": self.component_name,
-            "root_folder": self.root_folder,
+            "root_folder": str(self.root_folder),
```

Comment:
This is the fix.

Reply:
Yes, I was just waiting for user validation before finishing this PR. He shared this solved his bug. I will add a test and finish it next week.

Reply:
Done !

I discovered the bug when I found unprocessed key in the deep_diff.